core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala

```diff
@@ -116,6 +116,18 @@ abstract class AbstractConsumerTest extends BaseRequestTest {
     records
   }
 
+  protected def sendDuplicateRecords(producer: KafkaProducer[Array[Byte], Array[Byte]], numRecords: Int,
```

Comment:
How about renaming this to `sendRecordsWithKey` and include the key as a parameter? Otherwise it is a little strange as a general utility since it is tailored so narrowly for compaction. Also, perhaps we could consider moving this to `TestUtils`?

Reply:
I also thought it was a little out of place. I'll change it.